lib/listeners/base-listener.ts

```diff
@@ -85,7 +85,7 @@ export class BaseListener {
 
 		if (parts <= 3) routingKey = `${eventName}.${postFix}`;
 		else if (parts === 4) {
-			const baseName = `${split.splice(0, 3).join('.')}`;
+			const baseName = `${split.splice(0, 4).join('.')}`;
 			routingKey = `${baseName}.${postFix}`;
```

Comment:
I think this logic should be guaranteed on the event, also, `routingKey` is a concept of Rabbit and shouldn't be used outside it.